service/src/test/java/org/cbioportal/service/impl/ClinicalDataServiceImplTest.java

```diff
@@ -0,0 +1,249 @@
+                PATIENT_CLINICAL_DATA_TYPE);
+
+        Assert.assertEquals((Integer) 5, result.getTotalCount());
+    }
+
+    @Test
+    public void fetchClinicalDataSampleClinicalDataType() throws Exception {
+
+        List<String> studyIds = new ArrayList<>();
+        studyIds.add(STUDY_ID);
+        List<String> sampleIds = new ArrayList<>();
+        sampleIds.add(SAMPLE_ID);
+
+        List<SampleClinicalData> expectedSampleClinicalDataList = new ArrayList<>();
+        SampleClinicalData sampleClinicalData = new SampleClinicalData();
+        expectedSampleClinicalDataList.add(sampleClinicalData);
+
+        Mockito.when(clinicalDataRepository.getAllClinicalDataOfSampleInStudy(studyIds, sampleIds, ATTRIBUTE_ID,
+                PROJECTION, 0, 0, null, null)).thenReturn(expectedSampleClinicalDataList);
+
+        List<? extends ClinicalDataSummary> result = clinicalDataService.fetchClinicalData(studyIds, sampleIds,
+                ATTRIBUTE_ID, ClinicalDataServiceImpl.SAMPLE_CLINICAL_DATA_TYPE, PROJECTION);
+
+        Assert.assertEquals(expectedSampleClinicalDataList, result);
+    }
+
+    @Test
+    public void fetchClinicalDataPatientClinicalDataType() throws Exception {
+
+        List<String> studyIds = new ArrayList<>();
+        studyIds.add(STUDY_ID);
+        List<String> patientIds = new ArrayList<>();
+        patientIds.add(PATIENT_ID);
+
+        List<PatientClinicalData> expectedPatientClinicalDataList = new ArrayList<>();
+        PatientClinicalData patientClinicalData = new PatientClinicalData();
+        expectedPatientClinicalDataList.add(patientClinicalData);
+
+        Mockito.when(clinicalDataRepository.getAllClinicalDataOfPatientInStudy(studyIds, patientIds, ATTRIBUTE_ID,
+                PROJECTION, 0, 0, null, null)).thenReturn(expectedPatientClinicalDataList);
+
+        List<? extends ClinicalDataSummary> result = clinicalDataService.fetchClinicalData(studyIds, patientIds,
+                ATTRIBUTE_ID, PATIENT_CLINICAL_DATA_TYPE, PROJECTION);
+
+        Assert.assertEquals(expectedPatientClinicalDataList, result);
+    }
+
+    @Test
+    public void fetchMetaClinicalDataSampleClinicalDataType() throws Exception {
+
+        List<String> studyIds = new ArrayList<>();
+        studyIds.add(STUDY_ID);
+        List<String> sampleIds = new ArrayList<>();
+        sampleIds.add(SAMPLE_ID);
+
+        BaseMeta expectedBaseMeta = new BaseMeta();
+        expectedBaseMeta.setTotalCount(5);
+
+        Mockito.when(clinicalDataRepository.getMetaSampleClinicalData(studyIds, sampleIds, ATTRIBUTE_ID))
+                .thenReturn(expectedBaseMeta);
+
+        BaseMeta result = clinicalDataService.fetchMetaClinicalData(studyIds, sampleIds, ATTRIBUTE_ID,
+                ClinicalDataServiceImpl.SAMPLE_CLINICAL_DATA_TYPE);
+
+        Assert.assertEquals((Integer) 5, result.getTotalCount());
+    }
+
+    @Test
+    public void fetchMetaClinicalDataPatientClinicalDataType() throws Exception {
```

Comment:
These tests really cannot fail, but they are good placeholders for testing any added domain logic in the service layer code later.